src/main/scala/tapasco/filemgmt/ProgressTrackingFileWatcher.scala

```diff
@@ -0,0 +1,115 @@
+//
+// Copyright (C) 2019 ESA, TU Darmstadt
+//
+// This file is part of Tapasco (TPC).
+//
+// Tapasco is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Tapasco is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Tapasco.  If not, see <http://www.gnu.org/licenses/>.
+//
+package de.tu_darmstadt.cs.esa.tapasco.filemgmt
+
+import de.tu_darmstadt.cs.esa.tapasco.Logging._
+import de.tu_darmstadt.cs.esa.tapasco.util.Listener
+import MultiFileWatcher._
+import Events._
+import ProgressTrackingFileWatcher._
+
+import scala.concurrent.duration.Duration
+
+/**
+  * A [[MultiFileWatcher]] which tracks the overall logfile.
+  * Using a FSM, the current state of the Compose Progression is Determined and Logged
+  * to give the user an overview how far the compose-Task has progressed.
+  * @param _logger Optional logger instance to use.
+  * @param pollInterval Polling interval in ms (default: [[MultiFileWatcher.POLL_INTERVAL]]).
+  */
+class ProgressTrackingFileWatcher(_logger: Option[Logger] = None, pollInterval: Int = POLL_INTERVAL)
+  extends MultiFileWatcher(POLL_INTERVAL) {
+  private[this] final val logger = _logger getOrElse de.tu_darmstadt.cs.esa.tapasco.Logging.logger(getClass)
+
+  var currentState: Int = -1
+  val start: Long = System.currentTimeMillis()
+  var stageStart: Long = System.currentTimeMillis()
+  var lastLog: Long = 0
+
+  private lazy val listener = new Listener[Event] {
+    def update(e: MultiFileWatcher.Event): Unit = e match {
+      case LinesAdded(src, ls) => {
+        ls foreach {l =>
+          checkState(l)
+        }
+      }
+    }
+  }
+
+  /**
+    * Checks whether a transition has occured depending on the current line of the log.
+    * If so, the corresponding new state is logged.
+    * Additionally, the finished State is also logged with its runtime.
+    * @param string parsed String
+    */
+  private def checkState(string: String): Unit = {
+    val old = currentState
+    if(currentState == -1) {
+      currentState = currentState + 1
+    }
+    for(i <- progressionStringsInfo.indices) {
+      val progressionString = progressionStringsInfo(i)._1
+      if(i == -1 || (currentState == i - 1 && string.contains(progressionString))){
```

Comment:
Wouldn't you miss on some status information, if two stages are completed in a time interval shorter than the `POLL_INTERVAL`?

Reply:
If I understand the concepts of MultiFileWatcher correctly, the Log is checked every `POLL_INTERVAL` milliseconds. When new lines are added, these are checked in sequence, thus preventing a loss of status.